File: examples/agent-framework-webhooks/python/agent_webhooks/prompt.py
"""Safe prompt projection for inbound email."""

from e2a import AsyncInboundEmail

REPLY_INSTRUCTIONS = (
    "Reply helpfully and concisely to the email. Write 1-3 short paragraphs of "
    "body text only; do not include a Subject line or quote the original email."
)


def email_prompt(email: AsyncInboundEmail) -> str:
    """Project normalized facade fields into a framework-neutral prompt.

    Header and body values remain sender-controlled and must be treated as untrusted.
    """
    sender = email.from_ or "(missing)"
    verified = "yes" if email.verified else "no"
    flagged = "yes" if email.flagged else "no"
    return (
        f"From: {sender}\n"
        f"Subject: {email.subject}\n"
        f"Sender DMARC verified: {verified}\n"
        f"Policy flagged: {flagged}\n"
        f"\n"
        f"{email.text}"
    )

File: examples/agent-framework-webhooks/typescript/src/prompt.ts
import type { InboundEmail } from "@e2a/sdk/v1";

export const REPLY_INSTRUCTIONS =
  "Reply helpfully and concisely to the email. Write 1-3 short paragraphs of " +
  "body text only; do not include a Subject line or quote the original email.";

/**
 * Project normalized facade fields into a framework-neutral prompt.
 *
 * Header and body values remain sender-controlled and must be treated as
 * untrusted input.
 */
export function emailPrompt(email: InboundEmail): string {
  const sender = email.from ?? "(missing)";
  const verified = email.verified ? "yes" : "no";
  const flagged = email.flagged ? "yes" : "no";
  return (
    `From: ${sender}\n` +
    `Subject: ${email.subject}\n` +
    `Sender DMARC verified: ${verified}\n` +
    `Policy flagged: ${flagged}\n\n` +
    email.text
  );
}
